src/write_gcm.c

```diff
@@ -519,15 +519,19 @@ typedef struct {
   size_t size;
 } wg_curl_write_ctx_t;
 
-static size_t wg_curl_write_callback(char *ptr, size_t size, size_t nmemb,
+static size_t wg_curl_write_callback(void *ptr, size_t size, size_t nmemb,
                                      void *userdata);
 
-static int wg_curl_get_or_post(char *response_buffer,
-    size_t response_buffer_size, const char *url, const char *body,
-    const char **headers, int num_headers, _Bool silent_failures) {
+static int wg_curl_get_or_post(char **response, const char *url,
+    const char *body, const char **headers, int num_headers, 
+    _Bool silent_failures) {
   DEBUG("write_gcm: Doing %s request: url %s, body %s, num_headers %d",
         body == NULL ? "GET" : "POST",
         url, body, num_headers);
+  wg_curl_write_ctx_t write_ctx = {
```

Comment:
[Optional] This can go back to after we fill in the headers (just before `curl_easy_setopt(curl, CURLOPT_URL, url);`).